hack/verify-staging-client-go.sh

```diff
@@ -18,5 +18,73 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+V=""
+while getopts ":v" opt; do
+  case $opt in
+    v) # increase verbosity
+      V="-v"
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      ;;
+  esac
+done
+readonly V
+
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
-"${KUBE_ROOT}"/staging/copy.sh -v
+cd ${KUBE_ROOT}
+
+# Smoke test client-go examples
+go install ./staging/src/k8s.io/client-go/examples/...
+
+# Create a temporary GOPATH for apimachinery and client-go, copy the current HEAD into each and turn them
+# into a git repo. Then we can run copy.sh with this additional GOPATH. The Godeps.json will
+# have invalid git SHA1s, but it's good enough as a smoke test of copy.sh.
+if [ "${USE_TEMP_DIR:-1}" = 1 ]; then
+    TEMP_STAGING_GOPATH=$(mktemp -d -t verify-staging-client-go.XXXXX)
+    echo "Creating the temporary staging GOPATH directory: ${TEMP_STAGING_GOPATH}"
+    cleanup() {
+        if [ "${KEEP_TEMP_DIR:-0}" != 1 ]; then
+            rm -rf "${TEMP_STAGING_GOPATH}"
+        fi
+    }
+    trap cleanup EXIT SIGINT
+    mkdir -p "${TEMP_STAGING_GOPATH}/src/k8s.io"
+    ln -s "${PWD}" "${TEMP_STAGING_GOPATH}/src/k8s.io"
+else
+    TEMP_STAGING_GOPATH="${GOPATH}"
+fi
+for PACKAGE in apimachinery client-go; do
+    PACKAGE_PATH="${TEMP_STAGING_GOPATH}/src/k8s.io/${PACKAGE}"
+    echo "Creating a temporary ${PACKAGE} repo with a snapshot of HEAD"
+    mkdir -p "${PACKAGE_PATH}"
+    rsync -ax --delete staging/src/k8s.io/${PACKAGE}/ "${PACKAGE_PATH}/"
+    pushd "${PACKAGE_PATH}" >/dev/null
+    git init >/dev/null
+    git add *
+    git -c user.email="nobody@k8s.io" -c user.name="verify-staging-client-go.sh" commit -q -m "Snapshot"
+    popd >/dev/null
+done
+
+echo "Running godep restore"
+pushd "${TEMP_STAGING_GOPATH}/src/k8s.io/kubernetes" >/dev/null
+export GOPATH="${TEMP_STAGING_GOPATH}"
+godep restore ${V} 2>&1 | sed 's/^/  /'
+
+echo "Testing staging/copy.sh"
+staging/copy.sh -d 2>&1 | sed 's/^/  /'
+popd
+
+
+# Check that the apimachinery Godeps revision is actually in Github
+echo "Checking that a valid k8s.io/apimachinery is reference in client-go"
+APIMACHINERY_REV=$(jq '.Deps[]|select(.ImportPath|startswith("k8s.io/apimachinery/pkg/api/errors"))' staging/src/k8s.io/client-go/Godeps/Godeps.json | jq -r '.Rev')
+GITHUB_COMMIT_URL="https://github.com/kubernetes/apimachinery/commit/${APIMACHINERY_REV}"
+if ! curl -f -o /dev/null "${GITHUB_COMMIT_URL}"; then
+    echo "Failed to find referenced k8s.io/apimachinery commit ${APIMACHINERY_REV}."
+    echo
+    echo "NOTE: You cannot merge apimachinery changes and a resync'ed client-go in the same PR."
+    echo "      Give the k8s.io/apimachinery sync scripts a chance to run and reference the new"
+    echo "      revision in a new client-go."
```

Comment:
By "in a new client-go", do you mean in a new PR that only contains client-go changes?